test/jb_splits_store/set.test.js

```diff
@@ -0,0 +1,278 @@
+
+const ONE_DAY = 3600000;
+
+describe('JBSplitsStore::set(...)', function () {
+  const PROJECT_ID = 1;
+  const DOMAIN = 2;
+  const GROUP = 3;
+  let SET_SPLITS_PERMISSION_INDEX;
+
+  before(async function () {
+    let jbOperationsFactory = await ethers.getContractFactory('JBOperations');
+    let jbOperations = await jbOperationsFactory.deploy();
+    SET_SPLITS_PERMISSION_INDEX = await jbOperations.SET_SPLITS();
+  });
+
+  async function setup() {
+    let [deployer, projectOwner, ...addrs] = await ethers.getSigners();
+
+    let mockJbOperatorStore = await deployMockContract(deployer, jbOperatorStore.abi);
+    let mockJbProjects = await deployMockContract(deployer, jbProjects.abi);
+    let mockJbDirectory = await deployMockContract(deployer, jbDirectory.abi);
+
+    await mockJbOperatorStore.mock.hasPermission
+      .withArgs(projectOwner.address, projectOwner.address, PROJECT_ID, SET_SPLITS_PERMISSION_INDEX)
+      .returns(true);
+
+    await mockJbProjects.mock.ownerOf
+      .withArgs(PROJECT_ID)
+      .returns(projectOwner.address);
+
+    await mockJbDirectory.mock.controllerOf
+      .withArgs(PROJECT_ID)
+      .returns(projectOwner.address);
+
+    let jbSplitsStoreFact = await ethers.getContractFactory('JBSplitsStore');
+    let jbSplitsStore = await jbSplitsStoreFact.deploy(
+      mockJbOperatorStore.address,
+      mockJbProjects.address,
+      mockJbDirectory.address
+    );
+
+    let splits = makeSplits(addrs[0].address);
+
+    return { deployer, projectOwner, addrs, jbSplitsStore, splits, mockJbOperatorStore, mockJbProjects, mockJbDirectory };
+  }
+
+  function makeSplits(beneficiaryAddress, count=4) {
+    let splits = []
+    for (let i = 0; i < count; i++) {
+      splits.push({
+        preferClaimed: false,
+        percent: Math.floor(10000000 / count),
+        lockedUntil: 0,
+        beneficiary: beneficiaryAddress,
+        allocator: ethers.constants.AddressZero,
+        projectId: 0
+      });
+    }
+    return splits;
+  }
+
+  it('set(...) and corresponding events', async function () {
+    const { projectOwner, addrs, jbSplitsStore, splits, mockJbOperatorStore, mockJbDirectory } = await setup();
+
+    await mockJbOperatorStore.mock.hasPermission.returns(false);
+    await mockJbDirectory.mock.controllerOf.returns(addrs[0].address);
+
+    const tx = await jbSplitsStore.connect(projectOwner).set(
+      PROJECT_ID,
+      DOMAIN,
+      GROUP,
+      splits
+    );
+
+    // Expect one event per split in splits[]
+    await Promise.all(
+      splits.map(async (split, _) => {
+        await expect(tx)
+          .to.emit(jbSplitsStore, 'SetSplit')
+          .withArgs(PROJECT_ID, DOMAIN, GROUP, Object.values(split), projectOwner.address);
+      }),
+    );
+
+    // Get the current splits (for this proj/dom/group)
+    let splitsStored = await jbSplitsStore.splitsOf(PROJECT_ID, DOMAIN, GROUP);
+
+    //compare every currently stored splits to the one we've just sent
```

Comment:
Remove comments like this from the rest of the file if the code is self-documenting/not confusing. Less comments == better. They should be used only when very necessary.